### DISCUSSION TITLE
Render State

### DISCUSSION THREAD

**Discussion by workgroupengineering** [AvaloniaUI/Avalonia#16822] [Q&A]
I am developing a custom control that derives from Control and has a custom render logic. I would like to redraw only the modified parts between two iterations of Render. Do you have any suggestions on how to implement it? How can I save the previous state of the render?

**Comment by maxkatz6** [comment]
> I would like to redraw only the modified parts between two iterations of Render.

You can implement CompositionCustomVisualHandler, and invalidate only specific geometry pargs:
https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/CompositionPage.axaml.cs#L299
Note, it's going to be the most efficient only with some framebuffer renderers, like embedded devices.

> How can I save the previous state of the render?

Don't.
